src/metabase/related.clj

```diff
@@ -159,20 +168,21 @@
   [card]
   (->> (t2/select Card
          :table_id (:table_id card)
+         :type [:in [:model :question]]
          :archived false)
        filter-visible
        (rank-by-similarity card)
        (filter (comp pos? :similarity))))
 
-(defn- canonical-metric
+(defn- similar-metrics
   [card]
-  (->> (t2/select LegacyMetric
+  (->> (t2/select Card
          :table_id (:table_id card)
+         :type :metric
          :archived false)
        filter-visible
-       (m/find-first (comp #{(-> card :dataset_query :query :aggregation)}
-                           :aggregation
-                           :definition))))
+       (rank-by-similarity card)
+       (filter (comp pos? :similarity))))
```

Comment:
This is also a product question, IMO. I can imagine that now that metrics become first class citizens, we want to compare them against questions and models. E.g., people might write questions and models that are better modeled as metrics.